src/rust/engine/dep_inference/build.rs

```diff
@@ -155,23 +157,65 @@ pub trait Visitor {
     .unwrap();
 }
 
+fn gen_impl_hash_file(name: &'static str, source_dir: &Path, impl_dir: &Path, out_dir: &Path) {
+  let mut hasher = Sha256::default();
+  for entry in WalkDir::new(impl_dir)
```

Comment:
Do these give a consistent order across platforms and styles of checkout?

I imagine they probably don't? Maybe it's worth collecting the files into a vec and sorting by name?

Reply:
Very good thought. I'll make this deterministic